javalib/src/main/scala/java/lang/String.scala

```diff
@@ -216,23 +213,19 @@ final class _String()
     count - string.count
   }
 
-  def compareToIgnoreCase(string: _String): Int = {
+  def compareToIgnoreCase(str: _String): Int = {
     var o1 = offset
-    var o2 = string.offset
-    val end =
-      if (count < string.count) offset + count
-      else offset + string.count
+    var o2 = str.offset
+    val end = Math.min(offset + count, offset + str.count)
     while (o1 < end) {
-      val c1: Char = compareValue(value(o1))
-      val c2: Char = compareValue(string.value(o2))
+      val cmp = caseFold(this.charAt(o1)) - caseFold(str.charAt(o2))
```

Comment:
Here Scala.js uses `charAt` which on Native takes in account the offset and does bounds checking - should we avoid that and use `value(i)` directly?

Honestly, not sure how this passes as we are using the offsets incremented directly which I need to investigate.